lib/src/io_io.dart

```diff
@@ -43,7 +45,8 @@ Future<List<int>> readAsBytes(Uri uri) async {
     _throwIfFailed(response, uri);
     int length = response.contentLength;
     if (length < 0) length = 0;
-    var buffer = new Uint8Buffer(length);
+    // Create empty buffer with capacity matching contentLength.
+    var buffer = new Uint8Buffer(length)..length = 0;
     await for (var bytes in response) {
       buffer.addAll(bytes);
```

Comment:
That would require even more logic in case the `contentLength` wasn't correct. It probably is, but this way it works optimally when everything is correct, and still works if `contentLength` isn't available or is incorrect.

(It does suggest that a `new Uint8Buffer.capacity(length)` constructor might be useful).